src/os/bluestore/StupidAllocator.cc

```diff
@@ -139,10 +141,11 @@ int StupidAllocator::allocate(
     }
   }
 
-  // search down (origin)
+  // search down (from origin, and skip searched extents by hint)
```

Comment:
the allocator is allowed to return a short extent if it can't find a big one.. that's why it searches the smaller bins (in descending order).  'need_size' should probably be named 'want_size'.